### PR TITLE
window: supports HTML5 fullscreen api

### LINKED ISSUE #1314: HTML5 full-screen API is not implemented
Hi
I have embedded a couple of videos from youtube and vimeo in my app. 
But the fullscreen feature does not work - furthermore if the fullscreen button is clicked the scroll functionality of the app stops working until the page is refreshed..

-Tobias

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #1314

### CODE REVIEW

**Comment by deepak1556** [comment]
Does this need an api to determine if the current fullscreen is an HTML/renderer triggered ?

File: atom/browser/native_window.cc [hunk 1]
```diff
@@ -139,6 +139,8 @@ NativeWindow::NativeWindow(content::WebContents* web_contents,
   web_contents->SetDelegate(this);
   inspectable_web_contents()->SetDelegate(this);
 
+  native_window_.reset(this);
```
Comment (zcbenz) [inline atom/browser/native_window.cc]: Why do we store `this` pointer in a member of `this`?

Comment (deepak1556) [inline atom/browser/native_window.cc]: `IsFullscreenForTabOrPending` needs a const reference, any other approach i can take here ?

Comment (deepak1556) [inline atom/browser/native_window.cc]: using it [here](https://github.com/atom/electron/pull/1455/files#diff-5bdd7e81c152176a3e7488ced82afd1cR223)

Comment (zcbenz) [inline atom/browser/native_window.cc]: We can make `NativeWindow::IsFullscreen` a `const` method to solve it.

**Comment by deepak1556** [comment]
Fixed, good thing the api calls were changed a bit with `42` . Thanks!

**Comment by zcbenz** [comment]
:+1:

**Comment by jameslk** [comment]
Thank you!!

**Pull request merged**